### PR TITLE
Fix superfluous blank line; Fix warnings

### PR DESCRIPTION
**Pull request by okurz**
* Fix blank line on process output
* Fix undefined warnings on undefined 'ARCH'

### CODE REVIEW

File: osutils.pm [hunk 1]
```diff
@@ -100,7 +100,7 @@ sub _run {
     $p->channels(0)->quirkiness(1)->internal_pipes(0)->separate_err(0)->start;
     $p->on(stop => sub {
             while (defined(my $line = $p->getline)) {
-                $out .= $line;
+                chomp($out .= $line);
```
Comment (cfconrad) [inline osutils.pm]: Just curious, do we really need the output flatten to one line?

Comment (okurz) [reply osutils.pm]: well, every "line" should just be that, one line, isn't it?

Comment (cfconrad) [reply osutils.pm]: I think, output like
```
a
b
c
```
become
`abc`
instead of only stripping the last `\n`. If this is what you won't, ok.

Comment (Martchus) [reply osutils.pm]: Not according to my test:

```
  DB<1> $foo = "foo\nbar\nfoo\n";

  DB<2> print($foo);
foo
bar
foo

  DB<3> print(chomp($foo));
1
  DB<4> print($foo);
foo
bar
foo
  DB<5> 
```

Comment (okurz) [reply osutils.pm]: hm, because I also call chomp around the $out, could be. I should crosscheck this then.

Comment (cfconrad) [reply osutils.pm]: I agree what you are tested is what I would expect. But I think we have a different case, more like this:
```
open(my $f, '<', '/etc/os-release') || die("ERROR on read");
my $out = "";
while (my $line = readline($f)) {
        chomp($out .= $line);
}
print $out;
close($f);
```

Comment (cfconrad) [reply osutils.pm]: Maybe I have a misunderstanding from that `$p->getline` and it act different!?

Comment (Martchus) [reply osutils.pm]: @cfconrad I get your point - my example is not like this code. Then the code looks wrong, indeed.

Comment (okurz) [reply osutils.pm]: any suggestion what I should do instead?

Comment (cfconrad) [reply osutils.pm]: You could `chomp($out);` before return.

Comment (okurz) [reply osutils.pm]: yes, of course! Thanks :)

**Comment by okurz** [comment]
ok, so let's merge this first so that the tests in https://github.com/os-autoinst/os-autoinst/pull/1191 don't show warnings anymore :)

**Pull request merged**